src/fastSlow.js

```diff
@@ -0,0 +1,133 @@
+"use strict";
+
+const {
+  validateScoreFormat,
+  findIntervalIndex,
+  addressToAlias,
+} = require("./scores");
+
+// Some abbreviations.
+const add = (a, b) => a + b;
+const alias = (user) => addressToAlias(user.address);
+const cred = (user, intervalIndex) => user.intervalCred[intervalIndex];
+
+const createTotalCredMap = ({users}, intervalIndex) =>
```

Comment:
First off-- this is just a great example of where flow is helpful; seeing a type signature like `Map<UserAlias, number>` would help me orient on it. (Although in this case it would be `{[UserAlias]: number}`).

I've generally come to prefer ES6 maps over object maps because I find the semantics more predictable (e.g. never need to worry if you'll have a user named `__proto__`). So I would write this as:

```js
const sum = xs => xs.reduce(add, 0);

const createTotalCredMap = ({users}, intervalIndex) => {
  const aliasCredPair = u => [alias(u), sum(u.intervalCred.slice(0, intervalIndex + 1)]
  return new Map(users.map(userCredPair))
}
```

I think this is also more readable than the reduce-based code.

What do you think? I recognize there's a pitfall where reviewers orient on how they would have done it vs. how it needs to be done; I don't think there's a particular issue with your code, so would be OK merging it. But I think there's also benefit to making consistent calls across the codebase, e.g. let's always use Maps instead of object-maps.

Reply:
Yeah I think you're right, this has overall poor readability. I'm still in the process of working out how flow with NPM packaging will work nicely, doing that with widgets atm.

Still I would say, let's write these improvements down as the roadmap to a proper version 1 and publish a "it runs, but that's about it" to simultaneously find out what the other requirements are of using it for a notebook.

Reply:
Works for me.